Add Governance Document

mrocklin opened the pull request:
This takes and modifies the Jupyter document.  I recommend reviewing commit-by-commit.

Closes #2 #5

mrocklin committed:
Steal governance document from Jupyter

https://raw.githubusercontent.com/jupyter/governance/master/governance.md

https://github.com/jupyter/governance/blob/master/LICENSE.md

mrocklin committed:
Minor/cosmetic changes Jupyter -> Dask

mrocklin committed:
Council changes

mrocklin committed:
Rewrite Institutional Partner Section and add Community Partners

For institutional partners we include the option of direct funding and
services support.  We remove the distinction between contributors and
council members, and place the expectation of 1 FTE.

We create the notion of Community Partners

mrocklin committed:
swap BDFL and Steering Council in general prose

This isn't intended to have any significant change, but should convey
that we expect the Steering Council to make most decisions in practice.

mrocklin committed:
Bring partners under one heading

pentschev commented on governance.md:
```suggestion
The BDFL can appoint their successor, but it is expected that the Steering
```

pentschev commented on governance.md:
This isn't clear to me, is it supposed to be part of the final document or expected to be discussed and properly formulated as a statement (i.e., without a question/suggestion)?

mrocklin replied:
Yes, I felt similarly when reading it.  It appears in the original document from Jupyter so I left it in.  I'd be happy to turn it into a statement though.

pentschev replied:
IMHO, it makes more sense for this to be a statement. For people reading this, it may be very confusing to understand who's the question for.

mrocklin replied:
Done

pentschev replied:
LGTM. +1

pentschev commented on governance.md:
```suggestion
There are also many other Contributors listed in the logs of the various Dask
```

pentschev committed:
Apply suggestions from code review

Co-Authored-By: mrocklin <mrocklin@gmail.com>

mrocklin committed:
Rephrase Q&A as a statement, and simplify

mrocklin committed:
Merge branch 'governance' of github.com:mrocklin/governance into governance

pentschev commented:
Everything seems good in general, no further comments from me. :)

jacobtomlinson commented on governance.md:
```diff
@@ -83,32 +79,34 @@ will consist of a BDFL and Steering Council. We view this governance model as
 defer that authority to the consensus of the community discussion channels and
 the Steering Council (see below). It is expected, and in the past has been the
-case, that the BDFL will only rarely assert his/her final authority. Because
+case, that the BDFL will only rarely assert theirfinal authority. Because
```
`theirfinal` -> `their final`

mrocklin replied:
Fixed.  Thanks!

jacobtomlinson reviewed (commented):
Looks reasonable to me.

mrocklin committed:
fix typo

guillaumeeb commented on governance.md:
Why approached specificaly by the BDFL?

mrocklin replied:
Removed.  Thanks @guillaumeeb

guillaumeeb reviewed (commented):
On really small comment, but this looks fine to me!

mrocklin committed:
Remove "by the bdfl" in removal section

mrocklin commented:
If there are no further comments then I suggest that we merge this in 48 hours.  All current owners have been informed of this PR and the intent to merge.

martindurant commented:
+1

jakirkham commented on governance.md:
```diff
@@ -0,0 +1,340 @@
+The goal of The Project is to
+develop open source software and enable parallel computing within the PyData
+software ecosystem.  The Software developed
+by The Project is released under the BSD (or similar) open source license,
```
```suggestion
by The Project is released under the BSD-3-Clause (or similar) open source license,
```

jakirkham committed:
BSD -> BSD-3-clause

Co-Authored-By: mrocklin <mrocklin@gmail.com>

jakirkham commented on governance.md:
Would it be reasonable to simplify this to be an engaged and active community member? Asking as there may be good candidates that focus on specific areas that we would still want to recruit them to fill more of a leadership role (assuming they are interested of course 😉).

mrocklin replied:
What does it mean to be an engaged and active community member?

In my mind there is likely to be differing thoughts on what this means.  My intention behind this item is to make it clear that people need to be engaged in ways other than contributing code to solve their problems.  Having direction over the project as a whole requires understanding several code facets of the project (arrays, dataframes, scheduling), several facets of maintenance (code review, user issue handling, release management), and a history of service (solving other people's problems).

People may consider themselves engaged because they use Dask at work, and continuously push functionality that they need into the project.  In my mind this makes them an excellent contributor, but not yet eligible to be on a decision-making council.

However, as I re-read these words I recognize that it sounds like a council member must do all of these things and so, someone who never wrote documentation would be ineligible.  That was not my intent and I'd be happy to change things around.

Thoughts?

jakirkham replied:
That was generally what I was wondering when reading the governance text previously.

My impression is we will be able to recruit people that are strong in a few of these areas, but not all. So I think we would benefit by acknowledging we welcome a diversity of skillsets into council membership. Some simplification to that effect might be helpful.

TomAugspurger replied:
Note that the  sentence introducing this says that just *some* of the areas should be met, not all. Hopefully that address the concern of

> we will be able to recruit people that are strong in a few of these areas, but not all.

jakirkham replied:
I've pushed a commit that turns this into a list of qualities that are desirable while generally highlighting this individual needs to be involved in various activities. Have also hinted this list might not be comprehensive. Please take a look and let me know what you think. 🙂

jakirkham commented on governance.md:
Should we include a percentage threshold required to approve a member?  Is there a way to particular individuals that dissent?

Just to propose a potential solution here, we chose to use [Helios]( https://heliosvoting.org ) for anonymized voting in conda-forge (as some other projects had used it). This works well for us. Not to say that is the way we should go here, but it is one option.

mrocklin replied:
It's a good question to ask.  Historically this has been done with lazy-consensus.  Anyone can block, we leave a question open for a while and hope to get a few +1s.  We should probably formalize this though (or select some other approach).

jakirkham replied:
Thanks for formalizing it.

jakirkham commented on governance.md:
Could we list the acting steering council members here?

mrocklin replied:
If so then we'll have to keep this up to date.  I would only want to do this if someone volunteers to keep this maintained for the next year or two.

mrocklin replied:
Given that no one volunteered to do this I'm going to suggest that we not do this.

jakirkham replied:
Thus far it's not clear to me how much this list is expected to change over the next year or two. This is both important for determining the risk of it going stale as well as the amount of work any one individual will take on. Any thoughts?

TomAugspurger replied:
Do we want a one to one relationship between GitHub owners in the steering Council?

If not, we’ll be updating this as soon as the next member is added. We can move to an explicit list at that time?

jakirkham replied:
I had assumed that the steering council were also owners (and vice versa), but maybe that is not a reasonable assumption. How do you see these differing?

jakirkham replied:
I've gone ahead and pushed a change stating there is a 1-to-1 relationship of steering council members and owners. Please let me know your thoughts on this in a new diff for clarity (if that's alright).

jakirkham commented on governance.md:
Is there a percentage for this vote?

mrocklin replied:
I don't propose one, no.  Do you want to propose something?

jakirkham replied:
Please see [this comment]( https://github.com/dask/governance/pull/11/files#r277490283 ). To simplify things, would suggest we keep the discussion there until we've reached a decision.

jakirkham commented on governance.md:
Would be useful to have some percentage here for removing members as well.

mrocklin replied:
I would follow the standard system here.  If you have suggestions otherwise then please voice them.

jakirkham replied:
As there seem to be two threads about ejecting members, I'd suggest we consolidate them into one thread. Would arbitrarily suggest this one and we can apply whatever we decide for both cases (assuming that sounds agreeable).

For other decisions following the standard system (am guessing this is what is outlined in the new commit?) seems fine. When it comes to removing members though, I think the bar should be a bit higher. Would suggest a large majority (in conda-forge it's 75%). Also would suggest that not voting in this case does not count as approval given the nature of the decision.

While I've made this suggestion, would be happy to discuss further if there are other thoughts on this.

mrocklin replied:
> Would suggest a large majority (in conda-forge it's 75%). Also would suggest that not voting in this case does not count as approval given the nature of the decision.

Historically I haven't been able to get this kind of quorum among current organization owners.  I think that, at least with current engagement, this removes the possibility of ejecting people.

mrocklin replied:
I don't think that we should prescribe too much here.  The lazy consensus model seems to mostly work.  We can tweak things if necessary in the future.  I would prefer to see us add details only as necessary.  (though I acknowledge that, in cases like these, it's good to have foresight).

mrocklin replied:
> As there seem to be two threads about ejecting members, I'd suggest we consolidate them into one thread

Happy to consolidate things.  I'm not sure what the other thread is though.  Are you pointing to the paragraph just above this one about inactivity?

jakirkham replied:
> Happy to consolidate things. I'm not sure what the other thread is though. Are you pointing to the paragraph just above this one about inactivity?

Yes. Continuing here seems fine.

TomAugspurger replied:
I’m also fine with the lazy consensus.

jakirkham replied:
Matt and I talked about this offline. I'm going to write up some proposed text and push it to the PR so we can review together. Happy to iterate from there.

jakirkham replied:
I've pushed some text that provides some guidance on when this may be appropriate to use and when not. Please take a look and let me know your thoughts.

jakirkham replied:
Also have pushed some text borrowed from the membership document for additions and applied it to this removal section. Hopefully that strikes the balance between handling this carefully and not being overly complex. Please let me know if you have thoughts on that as well.

jakirkham commented on governance.md:
What sort of information are the potential subcommittees suppose to provide to support their case? Who is allowed to be a member of these subcommittees (core, contributors, etc.)? Is there a vote for creating the subcommittees? When are subcommittees dissolved (if ever)? Are there other existing subcommittees that should be listed here (e.g. CoC)? Where do the rosters for their members live? How should one contact a subcommittee about an issue?

mrocklin replied:
I don't have good answers for these (I'm mostly copying from the Jupyter doc here).  Is this something that you think we should get into?  If so, would you be interested in leading that process?

mrocklin replied:
Ping @jakirkham

jakirkham replied:
Was waiting to see if other members had thoughts on these questions before voicing my own.

My suggestion would be to drop the subcommittees section. It seems a bit too heavyweight for our needs today.

mrocklin replied:
Taking a look at this now.  I suspect that the original intent was to provide a structure for the NumFOCUS subcommittee in the next section.

jakirkham replied:
~~Looks like this was changed below. Marking as resolved. Please reopen if not.~~

Wrong diff. Reopening. Sorry for the noise.

TomAugspurger replied:
This being for numfocus makes sense.

jakirkham replied:
At least in conda-forge we use subcommittees to accelerate decision making on some focused issue by a small group of interested individuals that the larger group may not be tracking in detail. I'm not entirely sure how Jupyter uses them (outside of the NumFOCUS subcommittee).

jakirkham replied:
I've pushed some text to fill out some answers to the questions above. This follows roughly how subcommittees are used in conda-forge (e.g. to defer decisions to a select group of experts on an issue). Though am not attached to this model if there are other use cases we'd like to cover. Please take a look and share your thoughts.

jakirkham commented on governance.md:
What is the purpose of the 1 person external to the Steering Council?

mrocklin replied:
I don't know

jakirkham replied:
If the purpose is unknown, may I suggest we drop this line?

Edit: This may be irrelevant if we drop Subcommittees.

jakirkham replied:
Looks like this was changed below. Marking as resolved. Please reopen if not.

jakirkham reviewed (commented):
Generally this seems good. Thanks for getting the ball rolling on this Matt. Definitely not an easy task, but much appreciated. Have added some comments inline.

The main concern that came up a few time when reading this document was votes (particularly on contentious issues) currently lack a clear mechanism for making decisions. Namely where does this vote occur? How do we protect members of the minority/unpopular view from backlash? What thresholds are needed to make these decisions?

This point is important enough that I think we need to have some hard numbers and processes in place before proceeding. That said, there are communities that have solved these issues. So I think it is possible for us to solve them as well (and perhaps even borrow their solutions 😉).

There are some other points about visibility of members of different groups. Ways to contact them, etc. These would be good to have somewhere. Whether in this document or referenced to from this document. Though this is more of a technical point than a concern and am guessing this is generally agreeable. Happy to discuss though.

mrocklin commented:
Thanks for the careful review @jakirkham !

> The main concern that came up a few time when reading this document was votes (particularly on contentious issues) currently lack a clear mechanism for making decisions. Namely where does this vote occur? How do we protect members of the minority/unpopular view from backlash? What thresholds are needed to make these decisions?

Entirely agreed.  What are your thoughts on what we should do here?

Currently this happens informally over e-mail, with a lazy-consensus method.  To be the simplest approach would be to change this to a mailing list and spell out the lazy consensus method somewhere in this document.  Thoughts?

jakirkham commented:
Of course. Hope it was helpful.

That seems like a good start. For many things lazy consensus can be good and quick. Spelling that out somewhere and referencing it is useful.

In some cases where these issues get more focused subcommittees can be useful for just delegating the work to a smaller group (increasingly conda-forge does this and Jupyter does a bit too hence that text). IDK how important this is for us.

For some things that are more contentious, we may want to opt for a different metric for passing (unanimous, some large majority, etc.). It's also in these cases where it is useful to anonymize ballots.

mrocklin committed:
Add section on voting

mrocklin committed:
remove need for one external person in NumFOCUS subcommittee

mrocklin commented:
I would like to finish this process up soon.  @jakirkham can I ask you to take another look above and see if there are still issues above?  If so, could I ask you to take an active role in trying to address them?

TomAugspurger commented on governance.md:
```diff
@@ -0,0 +1,363 @@
+
+The Dask Project (The Project) is an open source software project.
+The goal of The Project is to
+develop open source software and enable parallel computing within the PyData
```
Change pydata to python? Dask.org uses “scale python”.

TomAugspurger commented on governance.md:
```diff
@@ -0,0 +1,363 @@
+Project consist of public websites and web-services that are hosted
+under the dask.org domain.
+
+The Project is developed by a team of distributed developers, called
```
Nitpick: change to “distributed team of developers” since we have a project named distributed :)

TomAugspurger commented on governance.md:
NumFOCUS capitalization.

TomAugspurger commented on governance.md:
Change Dask to The Project?  There are several of these in this section.

TomAugspurger reviewed (approved):
A couple minor comments.

mrocklin committed:
minor edits to governance document

mrocklin commented:
> Do we want a one to one relationship between GitHub owners in the steering Council?

Not necessarily. I could imagine people that we want governing the project that we don't want to have day-to-day github authority.  I don't have strong thoughts here and welcome edits.

> Note that the since introducing this says that just some of the areas should be not, not all.

I'm not sure I understand this

TomAugspurger commented:
Sorry, “since” should be sentence. I think that the requirement that just *some* of the items need be met should satisfy John’s concerns.

mrocklin commented:
As stated earlier, I would like to finish this up soon.  I appreciate people coming in and suggesting modifications.  That engagement is great.  However, if people are going to make suggestions, please make them *now* and then lets vote and be done.  I would prefer to not go through this process of making tiny adjustments for another month.  Also, while I put the original draft up here, I would really like it if people were to make edits directly, showing that this is a group effort.

TomAugspurger commented:
LGTM, no additional concerns. 

> On May 7, 2019, at 08:43, Matthew Rocklin <notifications@github.com> wrote:
> 
> As stated earlier, I would like to finish this up soon. I appreciate people coming in and suggesting modifications. That engagement is great. However, if people are going to make suggestions, please make them now and then lets vote and be done. I would prefer to not go through this process of making tiny adjustments for another month. Also, while I put the original draft up here, I would really like it if people were to make edits directly, showing that this is a group effort.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.

jrbourbeau reviewed (commented):
This LGTM, I'm +1. Thank you @mrocklin for pushing this through

TomAugspurger commented:
@jakirkham do you have time to summarize any outstanding concerns you have? Looking through, I *think* most have been addressed, but I may have missed some. If all is well, I suggest we merge in ~48 hours.

jakirkham commented:
@TomAugspurger, sorry I have a lot on my plate at work. If you are comfortable giving me until the end of the week, I can try to re-review and clearly summarize my concerns.

jakirkham committed:
Restructure council member requirements

Turns this longer item about the ideal steering council member into
a list of various activities the contributor may be engaged in that
help the broader community. Tries to balance the desire of indicating
this individual is deeply engaged in the community without requiring
they be involved in all facets.

jakirkham committed:
State 1-to-1 of steering council and owners

jakirkham committed:
Provide guidance on member removal

jakirkham committed:
Fill out details of subcommittee operation

jakirkham committed:
Specify conditions of subcommittee dissolution

jakirkham commented:
Alright I've pushed some changes to try and answer some of the open questions above. Am not too attached to these specific answers if there are other thoughts. Would be happy to iterate as needed. So please take a look and share your thoughts.

jakirkham committed:
Borrow text from member elections for removal

TomAugspurger commented on governance.md:
```diff
@@ -247,13 +247,28 @@ information that should not be posted to the public internet.
-the main private mailing list of the Council unless specifically called for.
+specific aspects of the project. Subcommittees are created using the standard
+voting procedure (e.g. lazy consensus). The members of the subcommittee need
+not all be Steering Council Members, but they must have at least one Steering
```
What's the rationale for requiring a steering council member? I think that if the steering council approves the formation of a subcommittee, we can trust that the leaders of the subcommittee will be responsible. We don't necessarily need a steering council member formally on the committee.

jakirkham replied:
Yeah this is stolen from conda-forge's governance (though we call them sub-teams). At least the reason in that context is that we wanted the ability to add non-core members to sub-teams. So this relaxes it from effectively requiring all steering council members to at least one on a subcommittee. Though I may be missing the question here.

TomAugspurger replied:
Gotcha. I was reading it from the other direction: why require *any* SC members.

But, this is minor. We can amend this later if this becomes an issue in practice.

jakirkham replied:
I see. That makes sense. Some thoughts on why this might be useful...

* A point of contact between the subcommittee and the steering council
* Some overall structure for where subcommittees fit in
* A BDFL delegate that the BDFL has easy access to

The last one is more specific to us. Perhaps there are other reasons. Thoughts?

TomAugspurger replied:
👍 Those are all good reasons.

mmccarty replied:
These are good points and give reasoning for the rule.  Should they be added?

TomAugspurger replied:
Personally, I'm comfortable with the rationale being here. If a future reader is interested, the discussion will be available.

jakirkham replied:
Thanks all. Am going to go ahead and mark this as resolved. Though please reopen if we missed something.

TomAugspurger commented on governance.md:
```diff
@@ -247,13 +247,28 @@ information that should not be posted to the public internet.
+subcommittee communications should happen on the main private mailing list of
+the Council unless specifically called for.
 
 The BDFL remains override authority over subcommittees, but should typically
```
Should "remains" be "retains"?

TomAugspurger commented on governance.md:
```diff
@@ -247,13 +247,28 @@ information that should not be posted to the public internet.
 
 The BDFL remains override authority over subcommittees, but should typically
-appoint a delegate who plays that role most of the time.
+appoint a delegate who plays that role most of the time. This delegate must
```
Same question: what's the rationale for requiring the BDFL-delegate be a steering council member?

jakirkham replied:
Let's keep the discussion in the thread above to avoid discussing it twice. We can update this text based on whatever consensus we reach.

jakirkham replied:
It sounds like we have resolved this above, @TomAugspurger. Am going to mark this one as resolved too. Though please feel free to reopen if we missed something.

TomAugspurger commented on governance.md:
What's the rationale for requiring a steering council member to be in the subcommittee?

jakirkham replied:
Let's keep the discussion in the thread above to avoid discussing it twice. We can update this text based on whatever consensus we reach.

TomAugspurger commented on governance.md:
Should "remains" be "retains"?

TomAugspurger commented on governance.md:
"e.g". -> "i.e." (or just remove)

TomAugspurger commented on governance.md:
"also be dissolved if they decide to be. " -> "choose to self-dissolve."

TomAugspurger commented on governance.md:
"to" -> "with"

mrocklin commented:
Thanks for the comments @TomAugspurger .  In many cases this text was just taken from the Jupyter govenrnance document, so there is no rationale on our side.

I encourage you to make changse directly that you think are unlikely to be contentious.

TomAugspurger commented:
I thought that the subcommittee-specific things were newly added by @jakirkham, but I may be wrong. I don't see anything in the Jupyter docs about it.

For now, I would prefer not requiring that a subcommittee be required to have a SC member, primarily because I don't know what subcommittees will be used for in practice. But I don't want to directly override @jakirkham's opinion if I don't understand the situation well enough :)

TomAugspurger committed:
style changes

jakirkham commented:
Thanks Tom. Yes, the subcommittee things are new. Let's leave that discussion in the one (arbitrarily chosen) thread above though to keep discussion easier to follow (if that's alright). Also thanks for the various typo fixes.

TomAugspurger commented:
It feels like this and https://github.com/dask/governance/pull/12 are close to reaching consensus. @jakirkham are you comfortable with calling a vote here as well?

jakirkham reviewed (approved):
Sure. Can everyone do a +1/-1? If new questions come up with the new changes, am happy to pause this. That said, let's plan on merging Monday if there are only +1s and no comments.

jakirkham commented:
...or Tuesday as I blanked and forgot Monday is a US Holiday. 😄

TomAugspurger commented:
Tuesday sounds good.

---

+1

mmccarty commented on governance.md:
How is "investments, employment, or contracting work, outside of The Project" defined in this context?  An example might be helpful.

jakirkham replied:
I think this is referring to people's day jobs. Though I may not fully understand the question.

mrocklin replied:
I think that in general yes, this most common case here is people's day jobs.  I think it's ok to leave this vague and subject to interpretation by the coucil though.

mmccarty commented on governance.md:
Not have a conflict of interest?

TomAugspurger replied:
Do you have any potential conflicts of interest with being a SC member in mind? I'm not sure what one would look like.

TomAugspurger replied:
Given the section below, it seems like having a conflict of interest isn't a problem as long as it's disclosed.

mmccarty replied:
Oh!  Then this could be "Maintains open disclosure of conflicts of interest."  This brings up one more comment below.

jakirkham replied:
That probably makes sense for new council members. Though I wouldn't really want to require it of arbitrary people contributing to the project.

mmccarty commented on governance.md:
Probably should add how members disclose conflicts. For example, "in writing as an issue or email".

jakirkham replied:
Thus far this has come up when discussing a particular piece of work usually during our stand-up meeting. Maybe that is good enough for now? What do other people think? @mrocklin @TomAugspurger @jrbourbeau (to pick on an arbitrary few :)?

mrocklin replied:
+1 to good enough for now

jakirkham replied:
Marking as resolved. Thank Mike!

mmccarty commented on governance.md:
Could there be cases where subcommittee members should not have access to the private mailing list of the Council?

jakirkham replied:
Yeah, I think you are right. Though this does create more management overhead as we would need multiple mailing lists. Any suggestions on how to keep the management overhead low here?

Edit: Should add part of my reason for agreeing is non-council members are allowed on subcommittees. So as they are non-council members they probably should not have access to all the same things council members have access to.

mrocklin replied:
I think it's fine to keep this unspecified.  The point here is that things should be public if possible, but not hidden from the council if they need to be private.  I don't want to get into a long discussion about structuring mailing lists.  At this point we're quite late in the review process.  I'd prefer to wrap things up.

jakirkham replied:
Thanks Matt. I agree with Mike there is something that would be good to resolve here, but also agree with you that this is getting into a finer grained conversation that shouldn't block integration of the governance document. Hopefully we can handle this discussion independently of integrating the governance document and update the document in the future based on what we decide. To that end, I've gone ahead and broken it out as issue ( https://github.com/dask/governance/issues/13 ) and will mark this thread as resolved.

mmccarty commented on governance.md:
"one fulltime maintainer"?  Probably should specify the number of months.  6 months?

jakirkham replied:
It's a good point. Why choose 6 months? Also wouldn't 6 months and full-time maintainer conflict?

mrocklin replied:
I think it's fine to keep things vague and at the discretion of the council.

Also, I don't want to open a new conversation here.  It's time to wrap this up.

jakirkham replied:
SGTM. @mmccarty, if this is a concern that you would like to follow-up on, could you please raise a new issue on this repo? If there is no need to follow-up on this further, let's go ahead and mark this as resolved. Thanks both for your engagement here.

mmccarty replied:
These are just things that I thought of while reading.  I agree with @mrocklin that its fine to keep it vague here.

jakirkham replied:
Cool, thanks for checking in. Will go ahead and mark this as resolved then.

mmccarty reviewed (commented):
Looks good to me.  Added some thoughts.

mmccarty commented on governance.md:
Assuming the conflict is found to be acceptable by the Council, how is the ruling documented?

jakirkham replied:
My reading of this text is council members are expected to be honorable members of the community. Namely if they feel their interest is conflicting, they voluntarily recuse themselves from voting. Does that make sense? Or is there a question about this that I've missed?

mmccarty replied:
Yes, that's fine by me.  Thanks for the clarification.

jrbourbeau commented:
+1

mrocklin commented:
+1

jcrist commented:
+1

quasiben commented:
+1

mmccarty commented:
+1

jakirkham commented:
Pinging all of the current owners one last time, @jakirkham @jcrist @jrbourbeau @martindurant @mrocklin @ogrisel @pitrou @shoyer @TomAugspurger. Planning on merging in 24-hours.

shoyer commented on governance.md:
Is this true yet? If not, maybe say "plans to affiliate" instead (or hold off on merging).

mrocklin replied:
Thoughts @aterrel ?

aterrel replied:
You are formally affiliated. You are not formally "sponsored" but the distinction is usually lost on folks.

shoyer replied:
Right, but at the end of this sentence it says "which serves as its fiscal sponsor". So that part should be updated.

aterrel replied:
Ah, I see now.

As worded the statement is still true.

NumFOCUS has three different types of projects. Fully fiscally sponsored, which means the steering council has all control over the assets that the project controls. Grantor/Grantee fiscal sponsorship, which means that the project has a grant that it controls the asset decisions per those given from the grant. And Affiliated projects, which is the nebulous "part of the club" project that has an account but all asset decisions must flow from the NumFOCUS board. While we use the term "fiscal sponsorship" most people will assume that all three types of projects are "fiscally sponsored". 

I think the Linux foundation would just call DASK "unfunded" but since we are a 501c3 and not a 501c6 we have to account who is making decisions with the assets.

jakirkham replied:
Does Andy's answer address your concern @shoyer or is there more to discuss/do here?

shoyer replied:
The bit about the NumFOCUS board having the ultimate decision power over the project still seems a bit weird to me with Dask's affiliate status, unless NumFOCUS is explicitly OK with this.

aterrel replied:
I didn't say ULTIMATE DECISION POWER _uses he-man voice_.

Just that the NumFOCUS board has decision over the assets, perhaps a better way to say this is, responsibility to ensure that funds are used for public good.

Anywho legal terms in a github comment are not conducive to full understanding of things, but effectively "affiliated" means we don't represent you as a fiscal entity legally. So a project is free to put assets under NumFOCUS and we will manage them to your benefit. We pretty much only do that for smaller things (such as the small grants program), because we realize that the NumFOCUS Board is ill equipped to make many decision with project's assets. 

Legally speaking though, we are a 501c3 which means any good that is owned by NumFOCUS for any reason, will have to be sold at market rate or donated to another 501c3. What "market rate" means can be left to some interpretation.

shoyer replied:
Try searching for "NumFOCUS" in this draft governance doc.

e.g., in this paragraph it says things like "NumFOCUS, which serves as its fiscal
sponsor, may hold project trademarks and other intellectual property, helps
manage project donations and acts as a parent legal entity"

This sounds like it's describing something more formal than is appropriate for an "affiliate" project.

aterrel replied:
Okay I was only reading the paragraph. 

For the most part, we can do that for any affiliated project, it's just that we aren't acting as the parent legal entity. That's for the full fiscal sponsorship, which one could wait to add to the doc.

As far as formality, there is a contract somewhere formalizing the affiliated project status between a group of Dask maintainers and the NumFOCUS org. I could probably get someone to dig it up if we need that for this document.

jakirkham replied:
It sounds like the text commented on in this thread is ok. Andy has followed up on the other parts in need of modification below. As such will mark this thread as resolved and handle the others after. Please let me know if I missed anything here though. 😉

shoyer reviewed (approved):
LGTM, aside from the one concern about whether dask is really part of NumFOCUS yet.

aterrel commented on governance.md:
There is only one NumFOCUS Board, so "Main NumFOCUS Board" is redundant. There may be a EU and US Board at some point but that hasn't really been established yet. #laywers

jakirkham replied:
IIRC this text was lifted from Jupyter's governance document. That said, agree it should be adjusted.

jakirkham replied:
Thanks for the suggestion. Dropped "Main".

aterrel commented on governance.md:
To be honest with you, I would rather we say NumFOCUS Board will oversee an election from a group of stakeholders.

jakirkham replied:
Yeah, it does feel a little strange. What would be the "group of stakeholders" in this case? The Steering Council? Something else?

jakirkham replied:
@aterrel, could you please clarify what you mean by "group of stakeholders"?

jakirkham replied:
Have pushed a commit to change this to the Steering Council will make this decision. Thanks for the feedback.

aterrel commented on governance.md:
You might add that this subcommittee is a requirement of fiscal sponsorship by NumFOCUS

jakirkham replied:
Sounds good.

jakirkham replied:
Thanks for the suggestion. Added some text to this effect.

pitrou commented on governance.md:
Should "consensus of current members" be "vote of current Council members"?

jakirkham replied:
Yeah, are you comfortable pushing this change?

jakirkham replied:
Thanks for the suggestion. Have changed the text accordingly.

pitrou commented on governance.md:
Hmm... what are "owners"?

jakirkham replied:
I guess owners of the org. Perhaps we should probably change it to steering council. Would you be comfortable pushing a change here, @pitrou?

jakirkham replied:
Thanks. Have changed to Steering Council members.

pitrou commented on governance.md:
So this section is called "voting" but it doesn't explain anything about voting actually, since voting seems to be eschewed.

jakirkham replied:
How do you mean?

jakirkham replied:
Any clarification here, @pitrou? As it stands, I'm inclined to leave this as-is.

mrocklin replied:
Calling this resolved due to lack of response.  (we're also past the stage where comments like this should slow things down).

pitrou commented:
This looks reasonable to me overall.

jakirkham committed:
Handle corrections raised by @pitrou

jakirkham committed:
Handle @aterrel's comments

jakirkham committed:
Remove a few more references to 'Main'

jakirkham commented:
Have pushed a few minor changes. @pitrou and @aterrel, if you are able to respond to the threads above, that would be very helpful.

jakirkham committed:
Delegate new BDFL decision to steering council

jakirkham commented:
@mrocklin and @aterrel, is this ok to merge?

TomAugspurger commented:
Thanks @jakirkham. +1 to [that change](https://github.com/dask/governance/pull/11/commits/4b98c5636361c3ea6c18f818005862fa1bbc1eab), and +1 in general.

mrocklin commented:
All comments have been resolved.  I plan to merge this tomorrow if there are no strong dissenting comments.

aterrel commented:
+1 I don't have a strong dissenting opinion. I can submit separate PRs on the language I would suggest.

jakirkham commented:
Thanks all! 😄 Please feel free to raise any additional things as issues and/or PRs. 😉
